flink-table-store-spark/src/main/java/org/apache/flink/table/store/spark/SparkRowData.java

```diff
@@ -0,0 +1,370 @@
+import org.apache.flink.table.data.DecimalData;
+import org.apache.flink.table.data.MapData;
+import org.apache.flink.table.data.RawValueData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.data.TimestampData;
+import org.apache.flink.table.store.utils.DateTimeUtils;
+import org.apache.flink.table.types.logical.ArrayType;
+import org.apache.flink.table.types.logical.DateType;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.MapType;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.types.RowKind;
+
+import org.apache.spark.sql.Row;
+
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/** A {@link RowData} wraps spark {@link Row}. */
+public class SparkRowData implements RowData {
+
+    private final RowType type;
+    private final Row row;
+
+    public SparkRowData(RowType type, Row row) {
+        this.type = type;
+        this.row = row;
+    }
+
+    @Override
+    public int getArity() {
+        return row.size();
+    }
+
+    @Override
+    public RowKind getRowKind() {
+        return RowKind.INSERT;
+    }
+
+    @Override
+    public void setRowKind(RowKind rowKind) {
+        if (rowKind == RowKind.INSERT) {
+            return;
+        }
+
+        throw new UnsupportedOperationException("Can not set row kind for this row.");
+    }
+
+    @Override
+    public boolean isNullAt(int i) {
+        return row.isNullAt(i);
+    }
+
+    @Override
+    public boolean getBoolean(int i) {
+        return row.getBoolean(i);
+    }
+
+    @Override
+    public byte getByte(int i) {
+        return row.getByte(i);
+    }
+
+    @Override
+    public short getShort(int i) {
+        return row.getShort(i);
+    }
+
+    @Override
+    public int getInt(int i) {
+        if (type.getTypeAt(i) instanceof DateType) {
+            return toFlinkDate(row.get(i));
+        }
+        return row.getInt(i);
+    }
+
+    @Override
+    public long getLong(int i) {
+        return row.getLong(i);
+    }
+
+    @Override
+    public float getFloat(int i) {
+        return row.getFloat(i);
+    }
+
+    @Override
+    public double getDouble(int i) {
+        return row.getDouble(i);
+    }
+
+    @Override
+    public StringData getString(int i) {
+        return StringData.fromString(row.getString(i));
+    }
+
+    @Override
+    public DecimalData getDecimal(int i, int precision, int scale) {
+        return DecimalData.fromBigDecimal(row.getDecimal(i), precision, scale);
+    }
+
+    @Override
+    public TimestampData getTimestamp(int i, int precision) {
+        return toFlinkTimestamp(row.get(i));
+    }
+
+    @Override
+    public <T> RawValueData<T> getRawValue(int i) {
+        throw new UnsupportedOperationException();
```

Comment:
Why not support the `getRawValue`? If this doesn't support, could you add the TODO to support in feature?

Reply:
add message: `Raw value is not supported in Spark, please use SQL types.`